src/Workspaces/CoreTest/CodeCleanup/ReduceTokenTests_WithDigitSeparators.cs

```diff
@@ -0,0 +1,2137 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeCleanup;
+using Microsoft.CodeAnalysis.CodeCleanup.Providers;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.UnitTests.CodeCleanup
+{
+    public class ReduceTokenTests_WithDigitSeparators
+    {
+        [Fact]
+        [WorkItem(5529, "DevDiv_Projects/Roslyn")]
+        [Trait(Traits.Feature, Traits.Features.ReduceTokens)]
+        [Trait(Traits.Feature, "WithDigitSeparators")]
+        public async Task ReduceSingleLiterals_LessThan8Digits()
+        {
+            var code = @"[|
+Module Program
+    Sub Main(args As String())
+        ' CATEGORY 1: Less than 8 significant digits
+        ' Dev11 and Roslyn behavior are identical: UNCHANGED
+
+        ' 5 significant digits
+        Const f_5_1 As Single = .149_95F         ' Dev11 & Roslyn: Pretty listed to 0.14995F
+        Const f_5_2 As Single = 0.149_95f        ' Dev11 & Roslyn: Unchanged
+        Const f_5_3 As Single = 1.499_5F         ' Dev11 & Roslyn: Unchanged
+        Const f_5_4 As Single = 149.95f          ' Dev11 & Roslyn: Unchanged
+        Const f_5_5 As Single = 1_499.5F         ' Dev11 & Roslyn: Unchanged
+        Const f_5_6 As Single = 14_995.0f        ' Dev11 & Roslyn: Unchanged
+
+        ' 7 significant digits
+        Const f_7_1 As Single = .149_999_5F      ' Dev11 & Roslyn: Pretty listed to 0.1499995F
+        Const f_7_2 As Single = 0.149_999_5f     ' Dev11 & Roslyn: Unchanged
+        Const f_7_3 As Single = 1.499_995F       ' Dev11 & Roslyn: Unchanged
+        Const f_7_4 As Single = 1_499.995f       ' Dev11 & Roslyn: Unchanged
+        Const f_7_5 As Single = 149_999.5F       ' Dev11 & Roslyn: Unchanged
+        Const f_7_6 As Single = 1_499_995.0f     ' Dev11 & Roslyn: Unchanged
+
+        Console.WriteLine(f_5_1)
```

Comment:
Why are any Console.WriteLine's necessary?

Reply:
@CyrusNajmabadi The `Console.WriteLine` is needed to ensure the variables are not discard in a release build.

Reply:
This is a test.  There is no actual build happening, right?  So why any concern about discarding?